src/main/java/org/mule/module/apikit/validation/BodyValidator.java

```diff
@@ -50,33 +50,31 @@ public static ValidBody validate(Action action, HttpRequestAttributes attributes
 
     ValidBody validBody = new ValidBody(payload);
 
-    if (action == null || !action.hasBody()) {
-      logger.debug("=== no body types defined: accepting any request content-type");
+    if (!action.hasBody()) {
       return validBody;
     }
 
-    final String requestMimeTypeName = getMediaType(attributes);
+    String requestMimeTypeName = getMediaType(attributes);
 
-    final Entry<String, MimeType> foundMimeType = action.getBody().entrySet().stream()
-        .peek(entry -> {
-          if (logger.isDebugEnabled())
-            logger.debug(format("comparing request media type %s with expected %s\n", requestMimeTypeName, entry.getKey()));
-        })
+    Entry<String, MimeType> foundMimeType = action.getBody().entrySet().stream()
         .filter(entry -> getMediaType(entry.getKey()).equals(requestMimeTypeName))
         .findFirst()
         .orElseThrow(UnsupportedMediaTypeException::new);
 
 
-    final MimeType mimeType = foundMimeType.getValue();
+    MimeType mimeType = foundMimeType.getValue();
 
-    if (requestMimeTypeName.contains("json") || requestMimeTypeName.contains("xml")) {
+    Object repeatableBody = makePayloadRepeatable(payload);
 
-      validBody = validateAsString(config, mimeType, action, requestMimeTypeName, payload, charset, errorTypeRepository);
+    if (requestMimeTypeName.contains("json") || requestMimeTypeName.contains("xml")) {
+      return validateAsString(config, mimeType, action, requestMimeTypeName, repeatableBody,
+                              charset != null ? charset : getCharset(attributes.getHeaders(), repeatableBody),
+                              errorTypeRepository);
 
     } else if ((requestMimeTypeName.contains("multipart/")
```

Comment:
Don't we have constants for "multipart" and "application/x-www-form-urlencoded"?